sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -96,6 +98,12 @@ trait CheckAnalysis extends PredicateHelper {
       case InsertIntoStatement(u: UnresolvedRelation, _, _, _, _) =>
         failAnalysis(s"Table not found: ${u.multipartIdentifier.quoted}")
 
+      case a: AlterTableStatement if isTempView(a.tableName) =>
+        a.failAnalysis("Cannot alter a view with ALTER TABLE. Please use ALTER VIEW instead.")
+
+      case s: StatementRequiringTable =>
```

Comment:
If `UnresolvedRelation` is used, then this isn't needed any more. I think that would also remove the need for `StatementRequiringTable`.